workshops/views.py

```diff
@@ -136,6 +140,58 @@ def person_details(request, person_id):
     return render(request, 'workshops/person.html', context)
 
 
+def person_bulk_add(request):
+    if request.method == 'POST':
+        form = PersonBulkAddForm(request.POST, request.FILES)
+        if form.is_valid():
+            try:
+                persons_tasks = _upload_person_task_csv(request, request.FILES['file'])
+            except csv.Error as e:
+                messages.add_message(request, messages.ERROR, "Error processing uploaded .CSV file: {}".format(e))
+            else:
+                for entry in persons_tasks:
```

Comment:
One thing I saw when testing your code a few days ago: we should have database transaction for this loop.

In case one entry in 1000 is wrong and the user has to fix it and then re-upload… they'll see database errors (unique constraint violation) due to re-adding previously added users / tasks.

Django transaction handling: https://docs.djangoproject.com/en/1.7/topics/db/transactions/#django.db.transaction.atomic